packages/api/src/mappings/google/activity.ts

```diff
@@ -40,23 +58,51 @@ export const mapToActivity = (
 
         if (data.dataSourceId === sourceIdActiveMinutes) {
           const seconds = convert(sum(intValues)).from("min").to("s");
-
           activity.summary = {
             ...activity.summary,
             durations: {
               ...activity.summary?.durations,
               active_seconds: seconds,
             },
           };
+          activity.activity_logs = activity.activity_logs?.map(act => {
+            const matchingActivity = data.point.find(point =>
+              matchActivityTime(act.start_time, act.end_time, parseInt(point.startTimeNanos))
+            );
+            if (matchingActivity) {
```

Comment:
nit: could put most of this in a reusable func that takes a callback function for how to modify the activity ie:

```
function addToActivityLogs(activity, data, callbackFn) {
    activity.activity_logs = activity.activity_logs?.map(act => {
            const matchingActivity = data.point.find(point =>
              matchActivityTime(act.start_time, act.end_time, parseInt(point.startTimeNanos))
            );
            if (matchingActivity) {
              act = callbackFn(act);
            }
            return act;
          });
}

```
and would be invoked like:

```
addToActivityLogs(activity, data,(act) => {
    return {
                ...act,
                durations: {
                  ...act.durations,
                  active_seconds: calculateActiveSeconds(act.start_time, act.end_time),
                },
              };
})
```

Reply:
I think in the current state, the logic between them is different enough to keep them separate. This should also make debugging easier.

Reply:
Not sure I follow - the callback would be different, but the wrapper is always the same?

I see this copy pasted in several locations:

```
    activity.activity_logs = activity.activity_logs?.map(act => {
            const matchingActivity = data.point.find(point =>
              matchActivityTime(act.start_time, act.end_time, parseInt(point.startTimeNanos))
            );
            if (matchingActivity) {
              act = callbackFn(act);
            }
            return act;
          });
```

not sure why it'd make debugging harder if you have the callback doing the specific logic as suggested